src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKey,
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKeyFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKey,
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKeyFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class DoNotGuardDictionaryRemoveByContainsKeyTests
```

Comment:
Could we add the following tests:

1. Negative condition on the `ContainsKey` (I know this looks stupid).

```csharp
if (!MyDictionary.ContainsKey(""Key""))
	MyDictionary.Remove(""Key"");
```

2. A test where the `ContainsKey` is combined with another bool check

```csharp
if (MyDictionary.ContainsKey(""Key"") && MyDictionary.Count > 2)
	MyDictionary.Remove(""Key"");
```

3. A test where the `ContainsKey` is set into a variable

```csharp
var result = MyDictionary.ContainsKey(""Key"");
if (result)
	MyDictionary.Remove(""Key"");
```

4. A test where the `Remove` is a line apart

```csharp
if (MyDictionary.ContainsKey(""Key""))
	var x = MyDictionary.Count;
	MyDictionary.Remove(""Key"");
```

Reply:
```csharp
if (MyDictionary.ContainsKey(""Key""))
	var x = MyDictionary.Count;
	MyDictionary.Remove(""Key"");
```

I had to go with `_ =` here, but other than that, I think I got these all covered now.